server/src-lib/Hasura/Tracing.hs

```diff
@@ -76,9 +79,10 @@ instance HasReporter m => HasReporter (ExceptT e m) where
 -- the active span within that trace, and the span's parent,
 -- unless the current span is the root.
 data TraceContext = TraceContext
-  { tcCurrentTrace  :: Word64
-  , tcCurrentSpan   :: Word64
-  , tcCurrentParent :: Maybe Word64
+  { tcCurrentTrace   :: BS.ByteString
```

Comment:
The encoding of the traced id's are in 16-character hex string. So should I keep the value as ByteString or change it to Text? 

IMO having the type as text would be better, since mentioning ByteString might mean that the filed can have values greater than UTF-8, which is not acceptable in request headers.

Reply:
As discussed, we think it's better to have the Tracing Context type to be a `Word64` since it is the lowest common denominator(_most common_) and create extra functions which will convert this Word64 into the format type we want to send.

For eg: We can have a `encodeHex` function to convert the `Word64` number to a 16 character hex string in the `injectHTTPContext` and`extractHTTPContext` functions.

This way we could support more formats in the future.